wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/Reactions/ReactionToggle.swift

```diff
@@ -0,0 +1,119 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireCommonComponents
+
+// MARK: - ReactionToggle
+
+class ReactionToggle: UIControl {
+
+    // MARK: - Properties
+
+    typealias ButtonColors = SemanticColors.Button
+
+    private let emojiLabel = DynamicFontLabel(fontSpec: .mediumRegularFont,
+                                              color: SemanticColors.Label.textDefault)
+    private let counterLabel = DynamicFontLabel(fontSpec: .mediumSemiboldFont,
+                                                color: SemanticColors.Label.textDefault)
+
+    private var onToggle: (() -> Void)?
+
+    var isToggled: Bool {
+        didSet {
+            guard oldValue != isToggled else { return }
+            updateAppearance()
+        }
+    }
+
+    // MARK: - Lifecycle
+
+    init(isToggled: Bool = false) {
+        self.isToggled = isToggled
+        super.init(frame: .zero)
+
+        layer.borderWidth = 1
+        layer.cornerRadius = 12
+        layer.masksToBounds = true
+
+        let stackView = UIStackView(arrangedSubviews: [emojiLabel, counterLabel])
+        stackView.axis = .horizontal
+        stackView.distribution = .fillEqually
+        stackView.alignment = .center
+        stackView.spacing = 4
+
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.layoutMargins = UIEdgeInsets(top: 4, left: 8, bottom: 4, right: 0)
```

Comment:
Why do you need this margin?

Reply:
For adding the proper spacing within the stack view where the emoji and counter lives.

Reply:
Why not to add this to main constrains in `NSLayoutConstraint.activate([ ...])`? I think it will be clearer.

Reply:
Why is the right insert 0 ?

Reply:
That's has been fixed!